File: 1932C.py
def get_prod(list):
    prod=1
    for i in range(len(list)):
        prod=prod*list[i]
    return prod

def solve():
    num = input()
    n,m = map(int, num.split())
    list=[]
    prod=1
    for _ in range(n):
        array = input()
        list.append(map(int,array.split()))
    s = input()
    left=0
    right=0
    print(prod&m,sep=' ')
    for i in range(n):
        if s[i]=='L':
            left+=1
            list = list[left:]
            prod = get_prod(list)
            print(prod%m,sep=' ')
        elif s[i]=='R':
            right+=1
            list = list[:right]
            prod = get_prod(list)
            print(prod%m,sep=' ')
    print()

t = int(input())
while(t>0):
    solve()

File: 1932C.cpp
#include <bits/stdc++.h>
using namespace std;
     
#define ll long long
     
void solve(){
    int n,m;cin>>n>>m;
    vector<ll>a(n);
    for(auto &e:a){cin >> e;}
    string s;cin >> s;
    vector<int>rem;
    int left=0,right=n-1;
    for(int i=0;i<n;i++){
        if(s[i]=='L'){
            rem.push_back(a[left++]);
            //left++;
        }else{
            rem.push_back(a[right--]);
            //right--;
        }
    }
    vector<ll>ans;
    ll prod=1;
    for(int i=n-1;i>=0;i--){
        prod = (prod*rem[i])%m;
        ans.push_back(prod);
    }
    for(int i=n-1;i>=0;i--){
        cout << ans[i] << " ";
    }
    cout << endl;
}    
     
int main(){
    int t;cin >> t;
    while(t--){
        solve();
    }
    return 0;
}